File: cc-app-android/app/src/main/java/org/caloriecloud/android/model/Challenge.java
package org.caloriecloud.android.model;

import java.io.Serializable;
import java.util.ArrayList;

public class Challenge implements Serializable{
    private int _id;
    private String type;
    private String name;
    private String description;
    private String startDate;
    private String endDate;
    private boolean fullSync;
    private String adminCode;
    private String participantCode;
    private String imageSrc;
    private int goal;
    private String impactMultiplier;
    private ArrayList<DeviceSetting> deviceSettings;
    private int defaultTeamId;
    private String notifyDate;
    private boolean notified;
    private boolean isActive;
    private int teamSelection;
    private String brand;
    private String color;
    private String createdAt;
    private String updatedAt;
    private int orgId;
    private int groupId;
    private Organization organization;
    private ArrayList<Team> teams;

    public Challenge(int _id, String adminCode, String brand, String color, String createdAt, int defaultTeamId, String description, ArrayList<DeviceSetting> deviceSettings, String endDate, boolean fullSync, int goal, int groupId, String imageSrc, String impactMultiplier, boolean isActive, String name, boolean notified, String notifyDate, Organization organization, int orgId, String participantCode, String startDate, ArrayList<Team> teams, int teamSelection, String type, String updatedAt) {
        this._id = _id;
        this.adminCode = adminCode;
        this.brand = brand;
        this.color = color;
        this.createdAt = createdAt;
        this.defaultTeamId = defaultTeamId;
        this.description = description;
        this.deviceSettings = deviceSettings;
        this.endDate = endDate;
        this.fullSync = fullSync;
        this.goal = goal;
        this.groupId = groupId;
        this.imageSrc = imageSrc;
        this.impactMultiplier = impactMultiplier;
        this.isActive = isActive;
        this.name = name;
        this.notified = notified;
        this.notifyDate = notifyDate;
        this.organization = organization;
        this.orgId = orgId;
        this.participantCode = participantCode;
        this.startDate = startDate;
        this.teams = teams;
        this.teamSelection = teamSelection;
        this.type = type;
        this.updatedAt = updatedAt;
    }

    public int get_id() {
        return _id;
    }

    public void set_id(int _id) {
        this._id = _id;
    }

    public String getAdminCode() {
        return adminCode;
    }

    public void setAdminCode(String adminCode) {
        this.adminCode = adminCode;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public int getDefaultTeamId() {
        return defaultTeamId;
    }

    public void setDefaultTeamId(int defaultTeamId) {
        this.defaultTeamId = defaultTeamId;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public ArrayList<DeviceSetting> getDeviceSettings() {
        return deviceSettings;
    }

    public void setDeviceSettings(ArrayList<DeviceSetting> deviceSettings) {
        this.deviceSettings = deviceSettings;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    public boolean isFullSync() {
        return fullSync;
    }

    public void setFullSync(boolean fullSync) {
        this.fullSync = fullSync;
    }

    public int getGoal() {
        return goal;
    }

    public void setGoal(int goal) {
        this.goal = goal;
    }

    public int getGroupId() {
        return groupId;
    }

    public void setGroupId(int groupId) {
        this.groupId = groupId;
    }

    public String getImageSrc() {
        return imageSrc;
    }

    public void setImageSrc(String imageSrc) {
        this.imageSrc = imageSrc;
    }

    public String getImpactMultiplier() {
        return impactMultiplier;
    }

    public void setImpactMultiplier(String impactMultiplier) {
        this.impactMultiplier = impactMultiplier;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isNotified() {
        return notified;
    }

    public void setNotified(boolean notified) {
        this.notified = notified;
    }

    public String getNotifyDate() {
        return notifyDate;
    }

    public void setNotifyDate(String notifyDate) {
        this.notifyDate = notifyDate;
    }

    public Organization getOrganization() {
        return organization;
    }

    public void setOrganization(Organization organization) {
        this.organization = organization;
    }

    public int getOrgId() {
        return orgId;
    }

    public void setOrgId(int orgId) {
        this.orgId = orgId;
    }

    public String getParticipantCode() {
        return participantCode;
    }

    public void setParticipantCode(String participantCode) {
        this.participantCode = participantCode;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public ArrayList<Team> getTeams() {
        return teams;
    }

    public void setTeams(ArrayList<Team> teams) {
        this.teams = teams;
    }

    public int getTeamSelection() {
        return teamSelection;
    }

    public void setTeamSelection(int teamSelection) {
        this.teamSelection = teamSelection;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }


}

File: cc-app-ios/Calorie Cloud/Challenge.swift
//
//  Challenge.swift
//  Calorie Cloud
//
//  Copyright © 2016 Calorie Cloud. All rights reserved.
//

import SwiftyJSON

class Challenge {
    
    var _id: Int!
    var type: String!
    var name: String!
    var description: String!
    var startDate: String!
    var endDate: String!
    var fullSync: Bool!
    var adminCode: String!
    var participantCode: String!
    var imageSrc: String!
    var goal: Int!
    var impactMultiplier: String!
    var deviceSettings: Array<DeviceSetting>!
    var defaultTeamId: Int!
    var notifyDate: String!
    var notified: Bool!
    var isActive: Bool!
    var teamSelection: Int!
    var brand: String!
    var color: String!
    var createdAt: String!
    var updatedAt: String!
    var orgId: Int!
    var groupId: Int!
    var organization: Organization!
    var teams: Array<Team>!
    
    init(_id: Int, type: String, name: String, description: String, startDate: String, endDate: String, fullSync: Bool, adminCode: String, participantCode: String, imageSrc: String, goal: Int, impactMultiplier: String, deviceSettings: Array<DeviceSetting>, defaultTeamId: Int, notifyDate: String, notified: Bool, isActive: Bool, teamSelection: Int, brand: String, color: String, createdAt: String, updatedAt: String, orgId: Int, groupId: Int, organization: Organization, teams: Array<Team>) {
        
        self._id = _id
        self.type = type
        self.name = name
        self.description = description
        self.startDate = startDate
        self.endDate = endDate
        self.fullSync = fullSync
        self.adminCode = adminCode
        self.participantCode = participantCode
        self.imageSrc = imageSrc
        self.goal = goal
        self.impactMultiplier = impactMultiplier
        self.deviceSettings = deviceSettings
        self.defaultTeamId = defaultTeamId
        self.notifyDate = notifyDate
        self.notified = notified
        self.isActive = isActive
        self.teamSelection = teamSelection
        self.brand = brand
        self.color = color
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.orgId = orgId
        self.groupId = groupId
        self.organization = organization
        self.teams = teams
        
    }
    
    init(json: JSON) {
        self._id = json["_id"].intValue
        self.type = json["type"].stringValue
        self.name = json["name"].stringValue
        self.description = json["description"].stringValue
        self.startDate = json["startDate"].stringValue
        self.endDate = json["endDate"].stringValue
        self.fullSync = json["fullSync"].boolValue
        self.adminCode = json["adminCode"].stringValue
        self.participantCode = json["participantCode"].stringValue
        self.imageSrc = json["imageSrc"].stringValue
        self.goal = json["goal"].intValue
        self.impactMultiplier = json["impactMultiplier"].stringValue
        self.deviceSettings = DeviceSetting.toArray(jsonArray: json["deviceSettings"].arrayValue)
        self.defaultTeamId = json["defaultTeamId"].intValue
        self.notifyDate = json["notifyDate"].stringValue
        self.notified = json["notified"].boolValue
        self.isActive = json["isActive"].boolValue
        self.teamSelection = json["teamSelection"].intValue
        self.brand = json["brand"].stringValue
        self.color = json["color"].stringValue
        self.createdAt = json["createdAt"].stringValue
        self.updatedAt = json["updatedAt"].stringValue
        self.orgId = json["orgId"].intValue
        self.groupId = json["groupId"].intValue
        self.organization = Organization(json: json["organization"])
        self.teams = Team.toArray(jsonArray: json["teams"].arrayValue)
        
    }
    

}
